### ISSUE TITLE
Use twisted.python.runtime in twisted.internet.serialport

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7176]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#7176|
|Type|enhancement|
|Created|2014-04-16 13:22:18Z|
|Branch|https://github.com/twisted/twisted/tree/serialport-platform-7176|

There's some platform checking code at the end of this module:

```
if os.name == 'posix':
    from twisted.internet._posixserialport import SerialPort
elif sys.platform == 'win32':
    from twisted.internet._win32serialport import SerialPort
```

This is silly.  Use `platform.isWindows()`.

Attachments:

* [7176-1.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a79/a793f807940238eea00d1aa91492b9b10476c45b/9c53b4d4b62018cbe8ef9bed50becefc0597ea6d.diff) (952 bytes) - added by adiroiban on 2014-04-22 07:44:38Z - 
* [7176-2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a79/a793f807940238eea00d1aa91492b9b10476c45b/473ba88bf321709ad4180f784a534ae8ee1cef25.diff) (1073 bytes) - added by adiroiban on 2014-04-23 12:01:56Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7176 7176
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_serialport_platform_7176 branches/serialport-platform-7176
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1397654538000000 1397654538000000
changetime__1415903488986599 1415903488986599
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban set owner to @adiroiban<br>@adiroiban set status to `assigned`|
|-|-|

I will try to fix this as exercise for submitting patches to twisted

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban removed owner<br>@adiroiban set status to `new`|
|-|-|

I have attach a first attempt to fix this ticket.


Please take a look. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to @adiroiban|
|-|-|

Thanks for your patch. I don't think introducing the `AssertionError` here is necessary.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

What should I do then for the else branch? ...

should I ignore it ?

should i just split into windows and non windows?

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Replying to [adiroiban](#note_4):
> What should I do then for the else branch? ...
> 
> should I ignore it ?

Yes it's currently ignored as well.

> 
> should i just split into windows and non windows?

Yes.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban removed owner|
|-|-|

I have attached a new patch.

Thanks for the review.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph commented|
|-|-|

(In [43455]) Branching to serialport-platform-7176.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph set owner to @glyph|
|-|-|

Aside from the pyflakes error, this looks fine to me.  I will address that and merge.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_9"></a>|@glyph commented|
|-|-|

(In [43457]) These imports are now unused and cause a pyflakes error.

re [#7176](https://github.com/twisted/twisted/issues/7176)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_10"></a>|@glyph set status to `closed`|
|-|-|

(In [43459]) Merge serialport-platform-7176: Use twisted.python.runtime to determine
platform in twisted.internet.serialport

Author: adiroiban

Reviewer: glyph

Fixes: [#7176](https://github.com/twisted/twisted/issues/7176)

This just replaces ad-hoc usage of os and sys modules with a supported
method of twisted.python.runtime.platform.
